### PR TITLE
Bump version to 8.0.0-dev

### PR DESCRIPTION
**Pull request by andrejtokarcik**
This PR results from updating the `VERSION` variable in `Makefile` and running `make version`.

The `master` branch already has [some code](https://github.com/gravitational/teleport/blob/01f88559e527484f2f2bbb6317ae9703a39218bc/lib/reversetunnel/srv.go#L1078-L1079) that assumes the most recent version belongs to v8 so this change is effectively necessary at this point in time.

### CODE REVIEW

File: examples/chart/teleport-cluster/Chart.yaml
Comment (r0mant) [inline examples/chart/teleport-cluster/Chart.yaml:4]: I think these should be bumped to `8.0.0-dev` as well, otherwise we might end up accidentally publishing 8.0 edge charts as v7. Same for kube-agent below. @webvictim Am I correct here or missing something?

Comment (webvictim) [reply examples/chart/teleport-cluster/Chart.yaml:4]: I think with the recent changes to chart publishing which now treat charts as artifacts (and are thus created by tag builds, rather than the old cron job) this isn't a risk as we only publish charts from `branch/v7`.

The reason I used to deliberately keep the version at `7` was because we don't publish `8.0.0-dev` Docker containers which means the chart won't deploy from `master` without modifications. See the logic here which special cases it: https://github.com/gravitational/teleport/blob/master/version.mk#L40-L45

Comment (webvictim) [reply examples/chart/teleport-cluster/Chart.yaml:4]: TL;DR This is expected and fine

**Review by webvictim — APPROVED** [review]

File: examples/chart/teleport-cluster/Chart.yaml
Comment (russjones) [reply examples/chart/teleport-cluster/Chart.yaml:4]: I think it's fine to update this, I just forgot to apply my change to `master`: f9ba4f2ec861d2f5e47c747bc217832a26276efa

@andrejtokarcik Can you add my commit to this PR as well then re-reun the version command to update this to `8`?

@webvictim The promotion pipeline creates docker containers for dev tags as far as I can tell? What am I missing? https://github.com/gravitational/teleport/blob/master/.drone.yml#L4160-L4197

Comment (andrejtokarcik) [reply examples/chart/teleport-cluster/Chart.yaml]: @russjones Done.

Comment (webvictim) [reply examples/chart/teleport-cluster/Chart.yaml]: @russjones We do indeed create Docker images for numbered `dev` versions (like `8.0.0-dev.3`) which have Git tags pushed and promoted.

We don't, however, push any tag without a numbered version though (like `8.0.0-dev` here) which would mean the Helm chart would fail to deploy.

**Review by Joerger — APPROVED** [review]

File: examples/chart/teleport-cluster/Chart.yaml
Comment (russjones) [reply examples/chart/teleport-cluster/Chart.yaml]: @andrejtokarcik @webvictim Okay, I think I got it. In that case, let's just make it `8.0.0-dev.1` and then afterwards cut a `8.0.0-dev.1` tag that way the Helm charts will work. Makes sense too as `master` is now Teleport 8 and we'll probably want to cut some dev, alpha, and beta releases from it.

Comment (andrejtokarcik) [reply examples/chart/teleport-cluster/Chart.yaml]: @russjones The Makefile implies that master is intended to be a dev version at all times, so alpha/beta should probably be cut from branch/v* not master, or the comment clarified:
https://github.com/gravitational/teleport/blob/e1b2a267a93f803a056f615c6e80efe7183c4bfb/Makefile#L9-L13

Nevertheless, if this PR is in a state acceptable to you, could you approve it?

Comment (webvictim) [reply examples/chart/teleport-cluster/Chart.yaml]: @andrejtokarcik is correct.

Comment (russjones) [reply examples/chart/teleport-cluster/Chart.yaml]: @andrejtokarcik The problem with cutting all dev, alpha, and beta builds off `branch/v8` will be that you guys will soon have to start merging to master than backport to `branch/v7` and `branch/v8` as well. Much easier from an operational perspective to just make `master` v8 right now.

That's what we did for Teleport 7 and it worked really well.

Comment (andrejtokarcik) [reply examples/chart/teleport-cluster/Chart.yaml]: I understand. Ideally the docs should then be updated to reflect the current process (also mentioning the needed numeric suffix for `dev` builds?). My sole interest here is to unblock #8150.

Comment (russjones) [reply examples/chart/teleport-cluster/Chart.yaml]: Yeah, approved. Make sure you cut `8.0.0-dev.1` tag and promote it in Drone after this PR is merged.

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**
